### PR TITLE
Django orm queue processors

### PR DESCRIPTION
**Pull request by ElliotAOram**
### Summary of work
* Removing old ORM aspects from QueueProcessor

### How to test your work
* Ensure this code is running on develop queue processor machine
* Manually submit runs (feel free to try via the webapp, via manual submission script, or run_detection)
* Ensure that runs are still successfully added to the database

* Code review
* Ensure unit tests pass

### Additional comments
* Note: The test coverage for the queueproc_utils and queue_processor will be handled in a separate PR. This PR is already large and there were not tests for these before so I don't intend to further inflate the PR.

### CODE REVIEW

**Comment by ElliotAOram** [comment]
This is a good revert point. In order to make further progress I think I need to remove ALL of the remain ORM instances from QP. If things go wrong, this is a good point to revert to

**Comment by ElliotAOram** [comment]
last class remaining is: `queue_processors/queue_processor/queueproc_utils/instrument_variable_utils.py`

File: queue_processors/queue_processor/queueproc_utils/tests/test_messaging_utils.py [hunk 1]
```diff
@@ -9,19 +9,110 @@
...
+        self.assertEqual(actual.run_number, self.run_number)
+        self.assertEqual(actual.instrument, self.instrument_name)
+        self.assertEqual(actual.rb_number, self.rb_number)
+        self.assertEqual(actual.data, file_path)
+        self.assertEqual(actual.reduction_script, script)
+        self.assertEqual(actual.reduction_arguments, args)
+        self.assertEqual(actual.run_version, self.version)
+        self.assertEqual(actual.facility, FACILITY)
+
+    @patch(UTILS_PATH + '.reduction_run_utils.ReductionRunUtils.get_script_and_arguments')
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queueproc_utils/tests/test_messaging_utils.py:101]: NIT to be more python3 if it works still should we change this to an fstring?
`f"{UTILS_PATH}.reduction_run_utils.ReductionRunUtils.get_script_and_arguments"`

File: queue_processors/queue_processor/queueproc_utils/tests/test_messaging_utils.py [hunk 2]
```diff
@@ -33,15 +124,17 @@ def test_send_pending_msg(self, mock_disconnect, mock_send, mock_connect, mock_c
...
+        test_data = "data_dict"
+        pending_queue_name = '/queue/ReductionPending'
+        MessagingUtils()._send_pending_msg(test_data)
 
         mock_client.assert_called_once()
         mock_connect.assert_called_once()
         mock_send.assert_called_once()
         mock_disconnect.assert_called_once()
 
         (args, kwargs) = mock_send.call_args
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queueproc_utils/tests/test_messaging_utils.py:136]: NIT, to align with the testing description docstring, should we assert the equality in a separate unit test?

File: queue_processors/queue_processor/queueproc_utils/tests/test_variable_utils.py [hunk 3]
```diff
@@ -8,89 +8,94 @@
...
-        expected = InstrumentJoin(name=self.valid_inst_var.name,
-                                  value=self.valid_inst_var.value,
-                                  is_advanced=self.valid_inst_var.is_advanced,
-                                  type=self.valid_inst_var.type,
-                                  help_text=self.valid_inst_var.help_text,
-                                  instrument=self.valid_inst_var.instrument,
-                                  experiment_reference=self.valid_inst_var.experiment_reference,
-                                  start_run=self.valid_inst_var.start_run,
-                                  tracks_script=self.valid_inst_var.tracks_script)
+        Test: An Variable object can is copied
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queueproc_utils/tests/test_variable_utils.py:85]: Minor:  Grammar revision to docstring needed

File: queue_processors/queue_processor/queueproc_utils/variable_utils.py [hunk 4]
```diff
@@ -47,15 +47,8 @@ def copy_variable(variable):
...
-                              help_text=variable.help_text,
-                              instrument=variable.instrument,
-                              experiment_reference=variable.experiment_reference,
-                              start_run=variable.start_run,
-                              tracks_script=variable.tracks_script)
+        variable.pk = None  # Creates a copy by changing primary key to None
+        return variable
 
     @staticmethod
     def get_type_string(value):
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queueproc_utils/variable_utils.py:54]: NIT (as not part of the code you have edited) Should we log if an unknown type is passed when returning text for unknown type inside `get_type_string?`

File: queue_processors/queue_processor/queueproc_utils/variable_utils.py [hunk 5]
```diff
@@ -24,21 +24,21 @@ class VariableUtils:
...
+        return model.RunVariable(name=instrument_var.name,
+                                 value=instrument_var.value,
+                                 is_advanced=instrument_var.is_advanced,
+                                 type=instrument_var.type,
+                                 help_text=instrument_var.help_text,
+                                 reduction_run=reduction_run)
 
     def save_run_variables(self, instrument_vars, reduction_run):
         """ Save reduction run variables in the database. """
         logger.info('Saving run variables for %s', str(reduction_run.run_number))
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queueproc_utils/variable_utils.py:37]: NIT convert to f string

File: queue_processors/queue_processor/queue_processor.py [hunk 6]
```diff
@@ -358,8 +321,7 @@ def reduction_skipped(self):
         reduction_run.reduction_log = self.message.reduction_log
         reduction_run.admin_log = self.message.admin_log
 
-        session.add(reduction_run)
-        session.commit()
+        db_access.save_record(reduction_run)
 
     def reduction_error(self):
```
Comment (JackEAllen) [inline queue_processors/queue_processor/queue_processor.py:326]: NIT This Method is a bit large. If can be done easily/cleanly, would be nice to break up into smaller methods

File: model/database/access.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    database = DatabaseClient()
+    database.connect()
+    return database
+
+
+def get_instrument(instrument_name, create=False):
+    """
+    Find the instrument record associated with the name provided in the database
+    :param instrument_name: The name of the instrument to search for
+    :param create: If True, then create the record if it does not exist
```
Comment (JackEAllen) [inline model/database/access.py:27]: NIT - Add Data types to params

File: model/database/access.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        save_record(instrument_record)
+    return instrument_record
+
+
+def get_status(status_value, create=False):
+    """
+    Find the status record associated with the value provided in the database
+    :param status_value: The value of the status record e.g. 'Completed'
+    :param create: If True, then create the record if it does not exist
+    :return: The Status object from the database
```
Comment (JackEAllen) [inline model/database/access.py:46]: NIT - Add data types to params

File: model/database/access.py [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        save_record(status_record)
+    return status_record
+
+
+def get_experiment(rb_number, create=False):
+    """
+    Find the Experiment record associated with the rb_number provided in the database
+    :param rb_number: The rb_number of the Experiment record e.g. 12345
+    :param create: If True, then create the record if it does not exist
+    :return: The Experiment object from the database
```
Comment (JackEAllen) [inline model/database/access.py:61]: NIT - Add data types to params

File: model/database/access.py [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        save_record(experiment_record)
+    return experiment_record
+
+
+def get_reduction_run(instrument, run_number):
+    """
+    Returns a QuerySet of all ReductionRun versions that have the given instrument/run_number
+    :param instrument: The name of the instrument to search for
+    :param run_number: The run number to search for
+    :return: A QuerySet of the ReductionRun records that match the criteria
```
Comment (JackEAllen) [inline model/database/access.py:77]: NIT - Add data types to params

File: queue_processors/queue_processor/queueproc_utils/variable_utils.py (hunk 4 above)
Comment (ElliotAOram) [reply queue_processors/queue_processor/queueproc_utils/variable_utils.py:54]: We control all these types via the database enum so I'm not too concerned plus the default fall through is to text which should handle any types we do not expect

File: queue_processors/queue_processor/queue_processor.py (hunk 6 above)
Comment (ElliotAOram) [reply queue_processors/queue_processor/queue_processor.py:326]: Agreed but not as part of this change. Will be done as part of the QueueProcessor re-write

**Pull request merged**